### PR TITLE
exclude future attendances from percentage present calculation

### PR DESCRIPTION
**Pull request by jshawl**

### CODE REVIEW

**Comment by nolds9** [comment]
:+1:

File: app/models/membership.rb [hunk 1]
```diff
@@ -46,6 +46,7 @@ def toggle_admin!
 
   def percent_from_status( association, status)
     assoc = self.send(association)
+    assoc = assoc.joins(:event).where("events.occurs_at < ? ", Time.now) if(association == :attendances)
```
Comment (adambray) [inline app/models/membership.rb]: @jshawl check out the [existing named scope](https://github.com/ga-dc/garnet/blob/master/app/models/attendance.rb#L9) for `due`!

I suspect you could use it here?

```
assoc = assoc.due if association == :attendances
```

File: spec/models/membership_spec.rb [hunk 2]
```diff
@@ -8,6 +8,9 @@
     @m.observations.create(status: 0)
     @m.observations.create(status: 0)
     @m.observations.create(status: 1)
+    cohort.events.create!(title:"Some test event2", occurs_at: Time.now)
+    @m.attendances.first.update(status:2)
```
Comment (adambray) [inline spec/models/membership_spec.rb]: You can use AR enum bang methods here after the other PR (#415) gets merged in:

```
@m.attendances.first.present!
```

File: app/models/membership.rb (hunk 1 above)
Comment (adambray) [inline app/models/membership.rb]: Related: My new years resolution is to never write anything more than a simple `where` clause on a model class outside of it's definition file.

**Comment by jshawl** [comment]
@adambray can haz merge? this is blocking pre-creating attendance events

**Comment by adambray** [comment]
:+1:

**Pull request merged**